routingv8/response.go

```diff
@@ -1,49 +1,36 @@
 package routingv8
 
 import (
-	"fmt"
-	"strconv"
+	"encoding/json"
 )
 
+type ErrorCodes []ErrorCode
+
+func (e *ErrorCodes) UnmarshalJSON(b []byte) error {
+	errorCodes := make([]int, 0)
+	if err := json.Unmarshal(b, &errorCodes); err != nil {
+		return err
+	}
+	codes := make([]ErrorCode, 0, len(errorCodes))
+	for _, errorCode := range errorCodes {
+		codes = append(codes, ErrorCode(errorCode))
+	}
+	*e = codes
+	return nil
+}
+
 type ErrorCode int
 
 // See https://developer.here.com/documentation/matrix-routing-api/8.6.0/api-reference-swagger.html
 // for detailed explanations of each error.
 const (
-	ErrorCodeUnspecified ErrorCode = iota
-	ErrorCodeSuccess
-	ErrorCodeDisconnected
-	ErrorCodeMatchingFailed
-	ErrorCodeParameterViolation
-	ErrorCodeUnknown
+	ErrorCodeSuccess            = 0
```

Comment:
If you don't specify that this is of ErrorCode then it will be just a normal int

Reply:
Not sure that I am following what you mean here?

Reply:
You haven't specified any type when defining the error codes. In Go a `const` is never typed until it is used.

`const ErrorCodeSuccess = 1` is not any type. But if you pass it as an argument to a function which takes an int64, for example, it then becomes an int64.

I noticed that there is a type ErrorCode defined

`type ErrorCode int` which I understand is meant to be the type of these error codes, right?

Because consts are not typed until used I think this is OK however, if you are working with a function that takes, for example, the `ErrorCode` type as an argument, you are not going to get any suggestions from your code editor for any of the defined consts because the type is only known at runtime. Making the type explicit here would help with suggestions.

It's not a blocker and maybe I missunderstood the purpose here so feel free to merge if it is not relevant

Reply:
Ah okay I see. Thanks for sharing. Good to know. I think the current solution works for now. Also added a test 👍